### PR TITLE
Reduce logging messages on connector shutdown.

### PR DESCRIPTION
**Pull request by boaks**
Signed-off-by: Achim Kraus <achim.kraus@bosch-si.com>

### CODE REVIEW

File: element-connector/src/main/java/org/eclipse/californium/elements/UDPConnector.java [hunk 1]
```diff
@@ -149,22 +150,22 @@ public synchronized void start() throws IOException {
 
 		// start receiver and sender threads
 		LOGGER.info("UDPConnector starts up {} sender threads and {} receiver threads",
-				new Object[]{senderCount, receiverCount});
+				new Object[] { senderCount, receiverCount });
```
Comment (sbernard31) [inline element-connector/src/main/java/org/eclipse/californium/elements/UDPConnector.java]: removing object[]

File: element-connector/src/main/java/org/eclipse/californium/elements/UDPConnector.java [hunk 2]
```diff
@@ -282,32 +287,39 @@ public void run() {
...
-			socket.receive(datagram);
-			LOGGER.debug("UDPConnector ({}) received {} bytes from {}:{}",
-					new Object[]{socket.getLocalSocketAddress(), datagram.getLength(),
-						datagram.getAddress(), datagram.getPort()});
-			byte[] bytes = Arrays.copyOfRange(datagram.getData(), datagram.getOffset(), datagram.getLength());
-			RawData msg = RawData.inbound(bytes, new AddressEndpointContext(datagram.getAddress(), datagram.getPort()), false);
-
-			receiver.receiveData(msg);
+			DatagramSocket socket;
+			synchronized (this) {
```
Comment (sbernard31) [inline element-connector/src/main/java/org/eclipse/californium/elements/UDPConnector.java]: What is the purpose of those synchronized block

File: element-connector/src/main/java/org/eclipse/californium/elements/UDPConnector.java [hunk 3]
```diff
@@ -319,28 +331,36 @@ private Sender(String name) {
...
+						datagram.getLength(), datagram.getAddress(), datagram.getPort());
+				raw.onError(new EndpointMismatchException());
+				return;
+			}
+			datagram.setData(raw.getBytes());
+			datagram.setAddress(raw.getAddress());
+			datagram.setPort(raw.getPort());
+
+			DatagramSocket socket;
+			synchronized (this) {
```
Comment (sbernard31) [inline element-connector/src/main/java/org/eclipse/californium/elements/UDPConnector.java]: What is the purpose of those synchronized block

File: element-connector/src/main/java/org/eclipse/californium/elements/UDPConnector.java (hunk 1 above)
Comment (boaks) [reply element-connector/src/main/java/org/eclipse/californium/elements/UDPConnector.java]: OK

File: element-connector/src/main/java/org/eclipse/californium/elements/UDPConnector.java (hunk 2 above)
Comment (boaks) [reply element-connector/src/main/java/org/eclipse/californium/elements/UDPConnector.java]: to ensure, that the synchronized start/stop has finished.

Comment (sbernard31) [reply element-connector/src/main/java/org/eclipse/californium/elements/UDPConnector.java]: :man_facepalming:  I missed that start and stop was synchronised method ...

**Comment by sbernard31** [comment]
LGTM

File: element-connector/src/main/java/org/eclipse/californium/elements/UDPConnector.java (hunk 2 above)
Comment (boaks) [reply element-connector/src/main/java/org/eclipse/californium/elements/UDPConnector.java]: I got in rare case NullPointerExceptiosn, therefore a touched the socket handling.

**Pull request merged**